apps/src/lib/ladle/delayed-component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  type ComponentType,
+  createElement,
+} from 'react';
+
+type DelayedComponentState = 'pending' | 'resolved';
+
+/**
+ * Creates a component wrapper that suspends for a specified delay.
+ *
+ * Useful for demonstrating Suspense behavior in Ladle stories without
+ * requiring actual dynamic imports or data fetching.
+ *
+ * @param Component - The component to wrap with artificial delay
+ * @param delayMs - Milliseconds to delay before resolving (default: 2000)
+ *
+ * @returns A component that throws a promise on first render, then resolves
+ *
+ * @example
+ * ```typescript
+ * const DelayedMyComponent = createDelayedComponent(MyComponent, 2000);
+ *
+ * export const WithSuspense = () => {
+ *   return createElement(
+ *     Suspense,
+ *     { fallback: createElement('div', null, 'Loading...') },
+ *     createElement(DelayedMyComponent, { someProp: 'value' }),
+ *   );
+ * };
+ * ```
+ */
+export const createDelayedComponent = <P extends object>(
```

Comment:
This should probably be named something like `lazyish`.  It's to mimic "lazy" loading ... other than about reducing bundle size and code splitting. It's to allow use of React's Suspense component and see progressive loading.